zeebe/gateway-protocol/src/main/proto/rest-api.yaml

```diff
@@ -627,6 +590,41 @@ paths:
         "500":
           $ref: "#/components/responses/InternalServerError"
 
+  /tenants/{tenantId}/users/search:
+    post:
+      tags:
+        - Tenant
+        - User
+      operationId: searchUsersForTenant
+      summary: Query users for tenant
+      description: Retrieves a filtered and sorted list of users for a specified tenant.
+      parameters:
+        - name: tenantId
+          in: path
+          required: true
+          description: The unique identifier of the tenant.
+          schema:
+            type: string
+      requestBody:
+        required: false
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/UserSearchQueryRequest"
+      responses:
+        "200":
+          description: The search result of users for the tenant.
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/UserSearchResult"
```

Comment:
❓ A little bit unrelated to the changes here because this endpoint is just reusing the user search result, but still relevant, I guess: The `UserResult` has an `id` attribute of type `int64` in this specification. However, looking at the [code that maps this](https://github.com/camunda/camunda/blob/main/zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/SearchQueryResponseMapper.java#L473-L479) to a REST DTO, the `id` is never set. Can the `id` be removed? Will this be filled in the future? If so, isn't this a `string` attribute?

Reply:
Yeah no idea where the `id` is coming from, AFAIK users never had an `id`.

Reply:
OK 😅